p2p/transport/quic/stream.go

```diff
@@ -19,16 +19,20 @@ type stream struct {
 var _ network.MuxedStream = &stream{}
 
 func (s *stream) Read(b []byte) (n int, err error) {
+	var streamErr *quic.StreamError
+
 	n, err = s.Stream.Read(b)
-	if err != nil && errors.Is(err, &quic.StreamError{}) {
+	if err != nil && errors.As(err, &streamErr) {
```

Comment:
I see, this is due to a breaking change in quic-go to make errors.Is be an approximation for an equality check instead of for type checking as it was before. Given the situation LGTM.

Reply:
That's correct. We implemented `errors.Is` incorrectly for the longest time, and now we finally fixed this.